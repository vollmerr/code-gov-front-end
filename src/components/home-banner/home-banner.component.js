/* global PUBLIC_PATH */

import React from 'react'

export default class HomeBanner extends React.Component {

  scrollToAbout() {
    const top = document.getElementById('banner-home').clientHeight;
    const offset = document.querySelector('header nav.main').clientHeight;
    window.scrollTo({
      top: top - offset,
      behavior: 'smooth'
    });
  }

  render() {
    const sectionStyle = {
      backgroundImage: `url('${this.props.backgroundImage}')`
    }
    return (
      <section id="banner-home" className="banner medium" style={sectionStyle}>
        <div className="banner-content">
          <div className="banner-title">{this.props.motto}</div>
          <div className="banner-subtitle show-w-gt-1200">{this.props.subtitle}</div>
        </div>
          <a
            className="scroll-indicator"
            title="Scroll Down"
            onClick={this.scrollToAbout}
          >
          <i className="icon icon-angle-down"></i>
        </a>

      </section>
    )
  }
}
