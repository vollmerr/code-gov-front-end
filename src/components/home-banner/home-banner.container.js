import { connect } from 'react-redux';
import { push } from 'connected-react-router'
import { getConfigValue, normalize } from 'utils/other'
import saveAgencies from 'actions/save-agencies'
import updateBrowseFilters from 'actions/update-browse-filters'
import HomeBannerComponent from './home-banner.component'

const mapStateToProps = ({ agencies }) => {
  return {
    agencies,
    backgroundImage: getConfigValue('images.background'),
    motto: getConfigValue('content.home.banner.motto'),
    subtitle: getConfigValue('content.home.banner.subtitle'),
    helpWantedTitle: getConfigValue('content.home.banner.help_wanted.title'),
    helpWantedDescription: getConfigValue('content.home.banner.help_wanted.description'),
    helpWantedButton: getConfigValue('content.home.banner.help_wanted.button'),
    issueUrl: getConfigValue('content.home.banner.issue_url'),
    browseByText: getConfigValue('content.home.banner.browse_by_text')
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onBrowseByEntityChange: event => {
      const value = normalize(event.target.value)
      if (value !== 'browse by agency') {
        let url = '/browse-projects'
        if (value !== 'all') {
          url += '?agencies=' + value
          dispatch(updateBrowseFilters('agencies', value, 'checked'))
        }
        dispatch(push(url))
      }
    },
    saveAgencies: () => dispatch(saveAgencies())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeBannerComponent)
