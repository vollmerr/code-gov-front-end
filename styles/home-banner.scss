#banner-home.banner {
  background-image: none;
  background-color: #0071bc; 
  background-size: auto; 
  background-repeat: no-repeat;
  background-position: 50% 50%;

  @media screen and (max-width: 988px) {
      margin-top: 50px;
  }

  @media screen and (max-width: 800px) {
    margin-top: 0px;
  }

  &.medium {
    height: 70vh;
  }

  .banner-content {
    position: absolute;
    top: 50%;

    .banner-subsection {
      height: 14rem;
    }
  }

  .banner-title {
    color: #ffffff;
    font-family: 'Source Sans Pro', sans-serif !important;
    font-size: 2em;
    margin-bottom: 0;
    margin-top: 0;
    text-transform: capitalize !important;
    font-weight: 900;
  
    @media screen and (min-width: $small-screen) {
      font-size: 2.8em;
    }
  
    @media screen and (min-width: $medium-screen) {
      font-size: 3.2em;
    }
  }

  .banner-subtitle {
    color: $color-white;
    font-family: 'Source Sans Pro', sans-serif !important;
    font-size: 24px !important;
    font-weight: 400;
    margin: 0 auto;
    max-width: 120rem !important;
  
    @media screen and (min-width: $small-screen) {
      font-size: 1.2em;
    }
  
    @media screen and (min-width: $medium-screen) {
      font-size: 1.4em;
    }
  }    
}
