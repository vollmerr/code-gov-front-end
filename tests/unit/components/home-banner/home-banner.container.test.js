import { mapStateToProps } from 'components/home-banner/home-banner.container'

const props = {
  agencies: [
    { acronym: 'a1', name: 'agency-1' },
    { acronym: 'a2', name: 'agency-2' },
  ],
}

describe('containers - HomeBanner', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps(props)).toMatchSnapshot()
    })
  })
})
