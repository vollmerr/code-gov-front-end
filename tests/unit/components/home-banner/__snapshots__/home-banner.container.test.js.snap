// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomeBanner mapStateToProps should return the correct properties 1`] = `
Object {
  "backgroundImage": "/assets/img/hero.png",
  "motto": "Coding California Government",
  "subtitle": "Code California is an open collaboration between agencies, industry technology partners and civic technologists working to code a more innovative, collaborative and effective government that best serves the people of California.",
}
`;
